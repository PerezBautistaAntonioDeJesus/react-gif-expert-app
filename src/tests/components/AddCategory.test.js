import { React } from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory"


describe('tests for <AddCategory /> component', () => {
    

    const setCategories = jest.fn();
    let wrapper =  shallow( <AddCategory setCategories={setCategories}/>);

    beforeEach(()=> {
        jest.clearAllMocks();
        wrapper =  shallow( <AddCategory setCategories={setCategories}/>);
    });

    test('should show the component correctly', () => { 
        
        expect( wrapper ).toMatchSnapshot();
     });


     test('should change the input value', () => { 
         const input = wrapper.find('input');

         input.simulate('change', { target: {
             value: 'Hola mundo'
         }})


         const p = wrapper.find('p');
         expect(p.text().trim()).toBe('Hola mundo')
      });


      test('shouldn`t show the information on submit', () => { 
          
        wrapper.find('form').simulate( 'submit', { preventDefault(){} }); 
        expect( setCategories ).not.toHaveBeenCalled();

       });

       test('should call setCategories function and clean the input', () => { 

        wrapper.find('input').simulate( 'change', { target: { value: 'Hola mundo' } } )

        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategories ).toHaveBeenCalled();

        expect( wrapper.find('input').prop('value') ).toBe( '' );

        })
})