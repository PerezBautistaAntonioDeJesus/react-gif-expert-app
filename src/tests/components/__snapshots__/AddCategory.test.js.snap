// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for <AddCategory /> component should show the component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    Add category
  </h2>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
